torch/_subclasses/fake_tensor.py

```diff
@@ -1070,7 +1070,9 @@ def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
         # NotImplemented lets us keep chaining until we find the actual
         # subclass
         cur_stack = _get_current_dispatch_mode_stack()
-        if fake_mode in cur_stack:
+        # NB: This must test for ANY fake tensor mode, because we want the
+        # fake tensor mode to take precedence
+        if any(isinstance(m, FakeTensorMode) for m in cur_stack):
             not_implemented_log.debug(
                 "FakeTensor mode already active: %s in %s", fake_mode, cur_stack
```

Comment:
nit: potentially inconsistent log when a `FakeTensorMode` other than `fake_mode` is in `cur_stack`.